### ISSUE TITLE
Alternative/Cheaper components

### ISSUE THREAD

**Issue by sebastianhodapp** [formtapez/ZigUP#3]
Dear @formtapez 

first of all: Kudos for this masterpiece of work! This versatile device looks like a perfect fit for so many possible applications!

When calculating and planning to build a batch myself, it looks like the relay and the PSU are really cost drivers. Have you had a look into alternative components?

For the PSU the HLK-PM03 could be a replacement with the same form factor, but less than 2,5$: https://www.mikrocontroller.net/attachment/349613/HLK-PM03.pdf

For the relay, I haven't found one wit the same dimensions (yet), but if the jqc3f-03vdc-c (https://voron.ua/files/pdf/relay/JQC-3F(T73).pdf) could be squeezed on the board, it would be a killer (from pricing perspective with <0,5$)

BR,
Sebastian

**Comment by formtapez** [comment]
Hi,
you are right, these two components are really not cheap.

I chose the Meanwell power supply for safety reasons. For something which is on mains 24/7 even when nobody is at home, i really dont like to have such HiLink module. But a dual-footprint on the pcb to still have the option would be a smarter idea.

The relay choice was not very clever, because i wanted a bistable one to safe standby-power. This in combination with only 3,3 Volt and a high switching current limited the selection possibilities on the market and drove the price up. When you try to squeeze another relais on the boards, you should take a bistable relais, too. For a normal monostable relais you have to patch the firmware.

I will make a improved board layout in some time in which these points get cleared.

**Comment by h4nc** [comment]
Would be great to make it as cheap as possible (while keeping it safe of course).

So first it would lower the cost if we could use parts that are available at aliexpress (no shipping cost and cheap prices). Most of the SMD are available there but I don't know if the the cable sockets are available too. The sonoff basic is a very low cost product. Maybe you use parts they are using (relay, power supply, should be cheap and save).

Another sugestions is to use the same pinheader as the CC2531 Sticks use. It's smaller and most zigbee2mqtt flashers have that cable. 
https://de.aliexpress.com/item/32972376101.html?spm=a2g0s.9042311.0.0.27424c4dKMNl5m

I hope I can get back to you when I have more ideas. Those are only suggestions.

Thanks!

**Comment by h4nc** [comment]
Another idea would be to make a second version that is able to dimm lights. 

Or maybe it’s possible to include that in one version. So it’s able to only switch on/off or dimmable.
There are Tuya dimmers out there that could help finding the right parts for that.

**Comment by sebastianhodapp** [comment]
> Hi,
> you are right, these two components are really not cheap.
> 
> I chose the Meanwell power supply for safety reasons. For something which is on mains 24/7 even when nobody is at home, i really dont like to have such HiLink module. But a dual-footprint on the pcb to still have the option would be a smarter idea.

As I would only install them into some garden lights, I wouldn't be concerned. But inside at home, makes totally sense.


> The relay choice was not very clever, because i wanted a bistable one to safe standby-power. This in combination with only 3,3 Volt and a high switching current limited the selection possibilities on the market and drove the price up. When you try to squeeze another relais on the boards, you should take a bistable relais, too. For a normal monostable relais you have to patch the firmware.

I must admit, I missed that detail in the datasheet. I like your design now even more, knowing that you though even about standby-power! But true. Limits the available products significantly. I'll keep an eye open.

> I will make a improved board layout in some time in which these points get cleared.

Wow, thanks!

**Comment by eddso** [comment]
First of all, very nice project, i like it, great work.
Here my suggestion for cheaper component relay:
https://lcsc.com/product-detail/Relays_HF163F-L-3-HL2T_C140724.html?currency=EUR
https://german.alibaba.com/product-detail/HF163F-L-3-HL2T-Pushbutton-Switches-60829718344.html

**Comment by formtapez** [comment]
I like your second single-coil relais. It is cheap and small (the first one is too high if you want to install the board behind a wallswitch).
But that relais only has switching capacity of up to 60 watt - That can be too weak if you still have some lightbulbs somewhere.

**Comment by h4nc** [comment]
@formtapez are you still planning to do a second version with cheaper components?

**Comment by formtapez** [comment]
@h4nc Yes, but it is not very high on my priority list :(

**Comment by martcs** [comment]
@formtapez do you still offer ZigUp pcb?  If yes than what cost it?

**Comment by formtapez** [comment]
@martcs yes. Use contact-info in last line of [README.md](https://github.com/formtapez/ZigUP/blob/master/README.md) and tell me destination country and amount of PCBs, so i can calculate a price.
